pokemon.go

```diff
@@ -78,6 +223,90 @@ func (p *Pokemon) HasValidEVs() bool {
 	return totalEVs <= 510
 }
 
+func (p *Pokemon) computeStats() {
+	if !p.HasValidLevel() {
+		log.Println(fmt.Printf("Failed to compute stats for %d, level %d invalid", p.NatDex, p.Level))
+	}
+
+	if !p.HasValidIVs() {
+		log.Println(fmt.Printf("Failed to compute stats for %d, ivs %d invalid", p.NatDex, p.IVs))
+	}
+
+	if !p.HasValidEVs() {
+		log.Println(fmt.Printf("Failed to compute stats for %d, evs %d invalid", p.NatDex, p.EVs))
+	}
+
+	// get base stats
+	pokemon_stats_csv := getCsvReader("data/pokemon_stats.csv")
+	var base_stats [6]uint
+	already_read_stats := 0
+	for {
+		record, err := pokemon_stats_csv.Read()
+		if p.NatDex == uint16(parseInt(record[0])) {
+			base_stats[parseInt(record[1])-1] = uint(parseInt(record[2]))
+			already_read_stats++
+		}
+
+		if already_read_stats == 6 {
+			break
+		}
+
+		if err == io.EOF {
+			break
+		}
+	}
+
+	// compute HP
+	hp_ev_term := math.Floor(float64(p.EVs[STAT_HP]) / 4)
+	base_iv_ev_level_hp_term := (2*base_stats[STAT_HP] + uint(p.IVs[STAT_HP]) + uint(hp_ev_term)) * uint(p.Level)
+	hp_floor_term := math.Floor(float64(base_iv_ev_level_hp_term) / 100)
+	p.Stats[STAT_HP] = uint(hp_floor_term + float64(p.Level) + 10)
+	p.CurrentHP = p.Stats[STAT_HP]
+
+	// compute all other stats
+	natureModifiers := p.Nature.getNatureModifers()
+	for _, stat := range [5]int{STAT_ATK, STAT_DEF, STAT_SPATK, STAT_SPDEF, STAT_SPD} {
+		ev_term := math.Floor(float64(p.EVs[stat]) / 4)
+		base_iv_ev_level_term := (2*base_stats[stat] + uint(p.IVs[stat]) + uint(ev_term)) * uint(p.Level)
+		floor_term := math.Floor(float64(base_iv_ev_level_term) / 100)
+		nature_term := float64(floor_term+5) * natureModifiers[stat]
+		p.Stats[stat] = uint(math.Floor(nature_term))
+	}
+}
+
+func (p *Pokemon) Equals(other Pokemon) bool {
```

Comment:
What's a scenario when we need to check if two Pokemon are equal, field-wise? You can have two Pokemon with the exact same stats, but we really should be checking pointers or index to see if they are the same.

Reply:
I have some tests for `GeneratePokemon()` set up that use `Equals()` to ensure its working properly. I guess I could change the test for the constructor's accuracy to only compare the stats, but how should I verify that `GeneratePokemon()` is working as expected without a method like this?

Reply:
Like you said, by the individual fields of the Pokemon. First write a test with no args and see if all the fields are their default value (level = 1, stats = 0 array, etc). Then, 1 parameter at a time you test that functionality: `GeneratePokemon(nature = adamant)` -> `if p.nature != adamant -> fail` and repeat.

Reply:
You should probably just be using `reflect.DeepEqual`.